plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -879,8 +884,33 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
 
         commit(appendFiles, session);
         transaction.commitTransaction();
+        // TODO this may not exactly be the snapshot we committed, if there is another writer
```

Comment:
Is there some action item? Looks like statement not a TODO. An issue?

Reply:
good point. I think https://github.com/trinodb/trino/issues/15439 would be the solution, so added link to that issue on this line too